pkg/functions/client.go

```diff
@@ -960,6 +966,178 @@ func (c *Client) Push(ctx context.Context, f Function) (Function, error) {
 	return f, nil
 }
 
+// ensureRunDataDir creates a .func directory at the given path, and
+// registers it as ignored in a .gitignore file.
+func ensureRunDataDir(root string) error {
```

Comment:
These functions support the functionality of the Client methods, so they were moved from function.go.  Keeping Function as closed to a data object as possible, with Client being the actor.